### PR TITLE
Fix #11576: Add duktapeConfig.cmake

### LINKED ISSUE #11576: No package 'duktape' found, CMake Error, A required package was not found, PKG_CHECK_MODULES
**OS:** macOS
**Version:** 10.14.6
**Commit/Build:** output of `brew info openrct2`

```
openrct2: stable 0.2.6 (bottled), HEAD
Open source re-implementation of RollerCoaster Tycoon 2
https://openrct2.io/
/usr/local/Cellar/openrct2/HEAD-c5d877f_1 (2,169 files, 35.4MB)
  Built from source on 2020-04-25 at 17:35:14
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/openrct2.rb
==> Dependencies
Build: cmake, pkg-config
Required: freetype, icu4c, jansson, libpng, libzip, openssl@1.1, sdl2, sdl2_ttf, speexdsp
==> Requirements
Required: macOS >= 10.13
==> Options
--HEAD
	Install HEAD version
```

When I upgrade OpenRCT2 using Homebrew:

`brew update && brew upgrade openrct2 --fetch-HEAD`

Then the output is:

```
Updating Homebrew...
==> Upgrading 1 outdated package:
openrct2 HEAD-c5d877f_1 -> HEAD-158adfb_1
==> Upgrading openrct2 HEAD-c5d877f_1 -> HEAD-158adfb_1 
==> Cloning https://github.com/OpenRCT2/OpenRCT2.git
Updating /Users/erik.huizinga/Library/Caches/Homebrew/openrct2--git
==> Checking out branch develop
Already on 'develop'
Your branch is up to date with 'origin/develop'.
HEAD is now at 158adfba8 Merge pull request #11560 from tupaschoal/more-coordz
==> Downloading https://github.com/OpenRCT2/title-sequences/releases/download/v0.1.2c/title-sequences.zip
Already downloaded: /Users/erik.huizinga/Library/Caches/Homebrew/downloads/0214e6a3fb336e8419adf432bd4fdee7ce81e7ceddcb87b4d48ac6e03643830f--title-sequences.zip
==> Downloading https://github.com/OpenRCT2/objects/releases/download/v1.0.14/objects.zip
Already downloaded: /Users/erik.huizinga/Library/Caches/Homebrew/downloads/2f140327558be78cd904098b1a6bde9608669066373af7b8e95e5890e4f6be98--objects.zip
==> cmake ..
Last 15 lines from /Users/erik.huizinga/Library/Logs/Homebrew/openrct2/01.cmake:


Google benchmark not found, disabling support
-- Checking for module 'duktape'
--   No package 'duktape' found
CMake Error at /usr/local/Cellar/cmake/3.17.2/share/cmake/Modules/FindPkgConfig.cmake:467 (message):
  A required package was not found
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.17.2/share/cmake/Modules/FindPkgConfig.cmake:647 (_pkg_check_modules_internal)
  src/openrct2/CMakeLists.txt:107 (PKG_CHECK_MODULES)
  CMakeLists.txt:267 (include)


-- Configuring incomplete, errors occurred!
See also "/tmp/openrct2-20200501-63097-wim1sf/build/CMakeFiles/CMakeOutput.log".

READ THIS: https://docs.brew.sh/Troubleshooting


==> Checking for dependents of upgraded formulae...
==> No dependents found!
```

And the file `"/tmp/openrct2...."` doesn't exist when the Homebrew process exits, so I don't have more logs.

What is this `duktape` module/package that seems missing?

### PR DESCRIPTION
**Pull request by AaronVanGeffen**
This PR adds a `duktapeConfig.cmake` script, allowing CMake to more easily find installed `duktape` libraries without path hints.

The primary reason for including this is that Homebrew paths were not being searched by default on macOS. This PR addresses this, and presumably this improves the locating process on other platforms as well.

### CODE REVIEW

File: src/openrct2/CMakeLists.txt [hunk 1]
```diff
@@ -101,17 +101,9 @@ if (NOT DISABLE_GOOGLE_BENCHMARK)
 endif ()
 
 if (ENABLE_SCRIPTING)
-    if (MSVC)
-        find_package(duktape REQUIRED)
-    else ()
-        PKG_CHECK_MODULES(DUKTAPE REQUIRED duktape)
-        target_include_directories(${PROJECT_NAME} PRIVATE ${DUKTAPE_INCLUDE_DIRS})
-        if (STATIC)
-            target_link_libraries(${PROJECT_NAME} ${DUKTAPE_STATIC_LIBRARIES})
```
Comment (AaronVanGeffen) [inline src/openrct2/CMakeLists.txt:110]: I'm not sure how the current changes influence making a static build with Duktape. As far as I can tell, I only have shared objects (.so) for Duktape on my machines (macOS/Homebrew and Arch Linux).

File: CMakeLists.txt [hunk 2]
```diff
@@ -11,6 +11,7 @@ include(CMakeDependentOption)
 
 if (NOT MSVC)
     include(FindPkgConfig)
+    set(CMAKE_PREFIX_PATH "${CMAKE_CURRENT_SOURCE_DIR}/cmake;${CMAKE_MODULE_PATH}")
```
Comment (janisozaur) [inline CMakeLists.txt:14]: Why is MSVC excluded here…

File: src/openrct2/CMakeLists.txt [hunk 3]
```diff
@@ -101,17 +101,9 @@ if (NOT DISABLE_GOOGLE_BENCHMARK)
...
-        else ()
-            target_link_libraries(${PROJECT_NAME} ${DUKTAPE_LIBRARIES})
-        endif ()
-    endif ()
+    find_package(duktape CONFIG REQUIRED)
```
Comment (janisozaur) [inline src/openrct2/CMakeLists.txt:104]: … when it can still require duktape here?

File: cmake/duktapeConfig.cmake [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# - Try to find duktape
```
Comment (janisozaur) [inline cmake/duktapeConfig.cmake:1]: There should be some information where does this file originate

File: CMakeLists.txt (hunk 2 above)
Comment (AaronVanGeffen) [reply CMakeLists.txt:14]: Because it's the CMake file from MSVC, as the commit description states.

File: src/openrct2/CMakeLists.txt (hunk 3 above)
Comment (AaronVanGeffen) [reply src/openrct2/CMakeLists.txt:104]: vcpkg provides the same duktapeConfig.cmake file in its ports.

File: cmake/duktapeConfig.cmake (hunk 4 above)
Comment (AaronVanGeffen) [reply cmake/duktapeConfig.cmake:1]: It says so in the commit description, but sure, I'll happily include a link to [duktapeConfig.cmake.in](https://github.com/microsoft/vcpkg/blob/master/ports/duktape/duktapeConfig.cmake.in) here as well.

File: src/openrct2/CMakeLists.txt (hunk 1 above)
Comment (AaronVanGeffen) [reply src/openrct2/CMakeLists.txt:110]: Should we at least expose the possibility to set `DUKTAPE_STATIC_LIBRARIES` manually for static builds? (Or indeed `LIBRARY`, singular.)

File: CMakeLists.txt (hunk 2 above)
Comment (janisozaur) [reply CMakeLists.txt:14]: Leave a comment in there so it's clear

File: cmake/duktapeConfig.cmake (hunk 4 above)
Comment (janisozaur) [reply cmake/duktapeConfig.cmake:1]: Please do

File: src/openrct2/CMakeLists.txt (hunk 1 above)
Comment (janisozaur) [reply src/openrct2/CMakeLists.txt:110]: I'll leave it up to you – if it's trivial enough to add, let's do this, but don't spend too much on it

**Pull request merged**